hydroflow_datalog_core/src/lib.rs

```diff
@@ -590,4 +614,55 @@ mod tests {
             "#
         );
     }
+
+    #[test]
+    fn test_choose() {
+        test_snapshots!(
+            r#"
+            .input clientIn
+            .output clientOut
+            .input clientAddr
+            
+            clientOut@addr(choose(i)) :~ clientIn(i), clientAddr(addr)
```

Comment:
Can we add runtime tests for this and count? Doesn't need to be a full Dedalus test, we can just run in Datalog + ticks single-node mode.